ee/desktop/user/server/server_darwin.go

```diff
@@ -27,3 +29,55 @@ func (s *UserServer) createSecureEnclaveKey(w http.ResponseWriter, _ *http.Reque
 	w.Write(keyBytes)
 	w.WriteHeader(http.StatusOK)
 }
+
+// getSecureEnclaveKey verifies that the public key exists in the secure enclave
+// then returns the public key back to follow REST conventions
+func (s *UserServer) getSecureEnclaveKey(w http.ResponseWriter, r *http.Request) {
+	b64 := r.URL.Query().Get("pub_key")
+	if b64 == "" {
+		http.Error(w, "pub_key is required", http.StatusBadRequest)
+		return
+	}
+
+	pubKey, err := echelper.PublicB64DerToEcdsaKey([]byte(b64))
+	if err != nil {
+		http.Error(w, fmt.Errorf("parsing pub_key: %w", err).Error(), http.StatusBadRequest)
+		return
+	}
+
+	// this verifies that the key exists in the secure enclave
+	signer, err := secureenclave.New(pubKey)
+	if err != nil {
+
+		// errKCItemNotFound = -25300
+		// means item was not found, any other error we assume is a different problem
+		// apple docs
+		// https://developer.apple.com/documentation/coreservices/1559994-anonymous/errkcitemnotfound/
+		// apple site where you can search for error codes, just enter error code
+		// https://developer.apple.com/bugreporter/
+		if strings.Contains(err.Error(), "-25300") {
```

Comment:
Dont really like the string contains .... but I don't think there is a way to pull "proper go errors" out of the apple C apis. At some point in the chain I think there would have to be a string or int compare and then creation of the go error.